File: core/src/net/bossdragon/component/render/Colorized.java
package net.bossdragon.component.render;

import com.artemis.PooledComponent;
import com.badlogic.gdx.graphics.Color;

/**
 * Color including transparency channel.
 */
public class Colorized extends PooledComponent {
    public final Color color = Color.WHITE.cpy();
    @Override
    protected void reset() {
        color.set(Color.WHITE);
    }
}

File: core/src/net/bossdragon/component/render/Colorized.kt
package net.bossdragon.component.render

import com.artemis.PooledComponent
import com.badlogic.gdx.graphics.Color

/**
 * Color including transparency channel.
 */
class Colorized : PooledComponent() {
    val color = Color.WHITE.cpy()

    override fun reset() {
        color.set(Color.WHITE)
    }
}
